app/Http/Controllers/Api/MovieController.php

```diff
@@ -75,12 +83,50 @@ public function destroy($id)
     /**
      * Show all movies for the current page.
      *
-     * @return \Illuminate\Http\Response
      * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
      */
     public function getCurrentPage(Request $request)
     {
         $size = $request->query('size');
         return $this->movieService->findCurrentPage($size);;
     }
+
+    /**
+     * Like functionality for current user.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function likeMovie($id)
+    {
+        $movie = $this->movieService->findByid($id);
+        switch ($movie) {
+            case $movie->getLikedByUserAttribute():
+                return $this->likeService->destroy(auth()->user(), $movie);
+            case $movie->getDislikedByUserAttribute():
+                return $this->likeService->dislikeIntoLike(auth()->user(), $movie);
+            default:
+                return $this->likeService->create(auth()->user(), $movie);
+        }
+    }
+
+    /**
+     * Dislike functionality for current user.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function dislikeMovie($id)
+    {
+        $movie = $this->movieService->findByid($id);
+        switch ($movie) {
+            case $movie->getLikedByUserAttribute():
```

Comment:
A ne znam, cudno mi je sa swithc-om, tj. ti radis swirtch da proveris da li je like-ovao ili nije, a `getLikedByUserAttribute` bas ne govori to iskreno :D

Reply:
A i ako je attribute mozes da pozoves: $movie->liked_by_user mislim svakako to menjas na scope :)